content/post/kubedb-v2024.11.18/index.md

```diff
@@ -0,0 +1,1269 @@
+  timeout: 5m
+  apply: IfReady # Apply only if the database is ready
+```
+
+Here, `new-custom-config` refers to the name of the custom configuration secret containing your desired SQL Server settings.
+
+Example 2: Applying Inline Configuration
+
+You can reconfigure using applyConfig like this:
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: MSSQLServerOpsRequest
+metadata:
+  name: msops-reconfigure-apply
+  namespace: demo
+spec:
+  type: Reconfigure
+  databaseRef:
+    name: mssqlserver-cluster
+  configuration:
+    applyConfig:
+      mssql.conf: |-
+        [memory]
+        memorylimitmb = 3072
+  timeout: 5m
+  apply: IfReady
+```
+
+In this example, the `applyConfig` field is used to specify the new configuration for the `mssql.conf` file directly in the Ops-Request manifest.
+
+**Reconfigure-TLS Ops-Request**
+
+The Reconfigure-TLS operation allows you to add, remove, or modify TLS configurations for an existing SQL Server cluster. It also supports advanced use cases like **rotating certificates** or **changing the certificate issuer**.
+
+Example 1: Adding TLS to a Cluster
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: MSSQLServerOpsRequest
+metadata:
+  name: msops-add-tls
+  namespace: demo
+spec:
+  type: ReconfigureTLS
+  databaseRef:
+    name: mssql-cluster
+  tls:
+    issuerRef:
+      name: mssqlserver-ca-issuer # Name of the cert-manager issuer
+      kind: Issuer # Issuer type (Issuer or ClusterIssuer)
+      apiGroup: "cert-manager.io"
+    certificates:
+      - alias: client
+        subject:
+          organizations:
+            - mssqlserver
+          organizationalUnits:
+            - client
+    clientTLS: true # Enable client TLS
+  timeout: 5m
+  apply: IfReady
+```
+
+This example demonstrates how to add TLS to an existing SQL Server cluster using certificates issued by `mssqlserver-ca-issuer`.
+
+Other Supported TLS Actions:
+
+**Remove TLS**: Disable TLS for the SQL Server cluster.
+
+tls Spec:
+```yaml
+tls:
+  remove: true # Remove existing TLS configuration
+```
+
+**Rotate Certificates**: Rotate the TLS certificates used by the SQL Server cluster.
+
+tls Spec:
+```yaml
+tls:
+  rotateCertificates: true # Trigger certificate rotation
+```
+
+**Change the Certificate Issuer**: Update the issuer of the TLS certificate to a new one.
+
+tls Spec:
+```yaml
+tls:
+  issuerRef:
+    name: mssqlserver-new-ca-issuer # New cert-manager issuer name
+    kind: Issuer # Issuer type (Issuer or ClusterIssuer)
+    apiGroup: "cert-manager.io" 
+```
+
+
+### Updates for MSSQLServer CRD
+
+**Removed Fields**:
+
+- `spec.internalAuth`: The `spec.internalAuth` section has been removed. TLS/SSL configurations for internal endpoint authentication of SQL Server Availability Group replicas should now be defined under `spec.tls`.
+
+**Updated Fields**:
+
+- Leader Election Configuration:
+  - The field `spec.leaderElection` is now moved under `spec.topology.availabilityGroup.leaderElection`.
```

Comment:
Leader Election Configuration: The field `spec.leaderElection` has been moved to `spec.topology.availabilityGroup.leaderElection`.